airbyte-integrations/connectors/source-clockify/source_clockify/streams.py

```diff
@@ -13,13 +13,17 @@
 
 
 class ClockifyStream(HttpStream, ABC):
-    url_base = "https://api.clockify.me/api/v1/"
+    url_base = ""
+    api_url = ""
+    api_path = "/api/v1/"
     page_size = 50
     page = 1
     primary_key = None
 
-    def __init__(self, workspace_id: str, **kwargs):
+    def __init__(self, workspace_id: str, api_url: str, **kwargs):
         super().__init__(**kwargs)
+        self.api_url = api_url
+        self.url_base = self.api_url + self.api_path
         self.workspace_id = workspace_id
 

```

Comment:
You can make this backward compatible if you transform the `url_base` to a property function in python.
Create a const to keep the previous base url.

        self.url_base = self.api_url + self.api_path
```suggestion
    def __init__(self, workspace_id: str, api_url: str, **kwargs):
        super().__init__(**kwargs)
        self.api_url = api_url or DEFAULT_API_URL
        self.workspace_id = workspace_id

    @property
    def url_base(self):
        return self.api_url + self.api_path
```

Reply:
Done!